### ISSUE TITLE
consider decoupling from log15

### ISSUE THREAD

**Issue by tmc** [kevinburke/rest#7]
folks may want to more easily bring their own logger.

**Comment by kevinburke** [comment]
You're right. Probably the best thing to do would be to declare a log interface here, have the loggers in this class conform to it, and also define a NullLogger or something.

**Comment by kevinburke** [comment]
Though depending on how smart the vendor tool in question is, it may still try to pull in log15.

Another option would be to split out the client into a subpackage, maybe "restclient", and then import that directly, which would avoid importing the dependency.
